pinot-core/src/main/java/org/apache/pinot/core/plan/TransformPlanNode.java

```diff
@@ -83,7 +83,11 @@ private void extractColumnsAndTransforms(BrokerRequest brokerRequest, IndexSegme
       // Extract selection expressions
       List<String> selectionColumns = selection.getSelectionColumns();
       if (selectionColumns.size() == 1 && selectionColumns.get(0).equals("*")) {
-        for (String column : indexSegment.getPhysicalColumnNames()) {
+        for (String column : indexSegment.getColumnNames()) {
```

Comment:
we can just get physical column names, right? Are we trying to say that virtual columns that do not start with dollar sign are ok?

Reply:
right. updated